### PR TITLE
Improve settings / help page, add Block Stubs

### PR DESCRIPTION
**Pull request by karloscarweber**
This PR combines Wordpress 5 help/settings page improvements with progress for new Gutenberg Blocks.

### CODE REVIEW

File: gumroad/css/admin.css [hunk 1]
```diff
@@ -1 +1,4 @@
 /* Completely Empty for now */
```
Comment (slavingia) [inline gumroad/css/admin.css]: probably want to remove the empty comment now :)

File: gumroad/scripts/gumblocks.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+(function( blocks, element ) {
+  const el = element.createElement;
+  const {TextControl} = wp.components;
+
+  import { TextControl } from '@wordpress/components';
+  import { withState } from '@wordpress/compose';
+
+  blocks.registerBlockType( 'gumroad/gumblocks', {
+    title: 'Gumroad Embed',
```
Comment (slavingia) [inline gumroad/scripts/gumblocks.js]: think we should support embed, overlay, and link. see this pr: #25 

will also effect how we do shortcodes. will elaborate more on that issue too.

File: gumroad/views/admin.php [hunk 3]
```diff
@@ -17,10 +17,63 @@
...
+
+	<p>With the Short Code Block you can use Short codes normally like you would in Wordpress <strong>4.9</strong>.</p>
+
+	<p>
+		<img style="width: 100%;max-width: 800px;" src="<?= plugins_url() ?>/gumroad/assets/images/gumroad-wp-003.png" alt="">
+	</p>
+
+	<h4>For wordpress 4.9.2 and lower:</h4>
+
+	<p>
```
Comment (slavingia) [inline gumroad/views/admin.php]: this paragraph seems to be a dupe of the one below.

File: gumroad/views/admin.php [hunk 4]
```diff
@@ -17,10 +17,63 @@
...
+
+	<!-- Add a Gumroad product help -->
+	<span style="margin-bottom:65px;display:block;" id="using_shortcodes"></span>
+	<h3>Using Shortcodes</h3>
+	<h4>For wordpress 5.0:</h4>
+
+	<p>Open the wordpress Editor and either type <code>/</code> or click on the <em>plus</em> icon and then choose to add a short code.</p>
+
+	<p>
+		<img style="width: 100%;max-width: 800px;" src="<?= plugins_url() ?>/gumroad/assets/images/gumroad-wp-001.png" alt="">
```
Comment (slavingia) [inline gumroad/views/admin.php]: you can move this style stuff to the css file. 

```
img {
   width: 100%; max-width: 800px:
}
```

File: gumroad/css/admin.css [hunk 5]
```diff
@@ -1 +1,4 @@
...
+.gumroad-wrap {
+  max-width:800px;
```
Comment (slavingia) [inline gumroad/css/admin.css]: space after ":"

File: gumroad/views/admin.php [hunk 6]
```diff
@@ -17,10 +17,63 @@
 }
 ?>
 
-<div class="wrap">
-
+<div class="wrap gumroad-wrap">
 	<h2><?php echo esc_html( get_admin_page_title() ); ?></h2>
 
+	<p>The Gumroad plugin makes it easy to use wordpress short codes to add links and embedded
```
Comment (slavingia) [inline gumroad/views/admin.php]: this paragraph can go. no need for marketing copy in this section.

File: gumroad/views/admin.php [hunk 7]
```diff
@@ -17,10 +17,63 @@
...
+products from your Gumroad storefront directly into your wordpress pages and posts. There is
+a brand new post and page editor in Version 5.0 of wordpress and onward, Gumroad links and
+embed code will work the same, but only if they are added properly.</p>
+
+	<span style="margin-bottom:65px;display:block;"></span>
```
Comment (slavingia) [inline gumroad/views/admin.php]: this can go too. we should rely as much on native styling as possible.

File: gumroad/views/admin.php [hunk 8]
```diff
@@ -17,10 +17,63 @@
...
+	<h3 class="title"><?php esc_html_e( 'Important SSL Requirement', 'gum' ); ?></h3>
```
Comment (slavingia) [inline gumroad/views/admin.php]: you can remove this too.

File: gumroad/views/admin.php [hunk 9]
```diff
@@ -17,10 +17,63 @@
...
+
+	<p>
```
Comment (slavingia) [inline gumroad/views/admin.php]: remove everything on here except for the new blocks documentation.

File: gumroad/views/admin.php [hunk 10]
```diff
@@ -17,10 +17,63 @@
...
+				'so you\'ll need to setup an SSL certificate for your site if you haven\'t already.', 'gum' ); ?>
+		</b></em>
+	</p>
+
+	<p>
+		If you need help setting up HTTPS on your Wordpress site here is <a href="http://wordpress.org/plugins/wordpress-https/" target="_blank">a Wordpress plugin</a> that can help.
+	</p>
+
+	<p>
+		<?php esc_html_e( 'Please refer to the Overlay and Embed documentation for additional help:', 'gum' ); ?>
```
Comment (slavingia) [inline gumroad/views/admin.php]: also remove all instances of esc_html_e – we don't use that anymore to keep thigns simple.

File: gumroad/views/admin.php [hunk 11]
```diff
@@ -17,10 +17,63 @@
...
+	<p>
+		<a href="#using_shortcodes">Using Shortcodes</a>, <a href="https://gumroad.com/embed" target="_blank">Embed Documentation</a>
+	</p>
+
+	<!-- Add a Gumroad product help -->
+	<span style="margin-bottom:65px;display:block;" id="using_shortcodes"></span>
+	<h3>Using Shortcodes</h3>
+	<h4>For wordpress 5.0:</h4>
+
+	<p>Open the wordpress Editor and either type <code>/</code> or click on the <em>plus</em> icon and then choose to add a short code.</p>
```
Comment (slavingia) [inline gumroad/views/admin.php]: it's WordPress! :)

update all of these that are left

File: gumroad/views/admin.php [hunk 12]
```diff
@@ -17,10 +17,63 @@
...
+
+	<p>
+		<img style="width: 100%;max-width: 800px;" src="<?= plugins_url() ?>/gumroad/assets/images/gumroad-wp-001.png" alt="">
+	</p>
+
+	<p>With the Short Code Block you can use Short codes normally like you would in Wordpress <strong>4.9</strong>.</p>
```
Comment (slavingia) [inline gumroad/views/admin.php]: They're shortcode blocks. whenever we are referring to WP or its features make sure to use the proper spacing/capitalization etc.

File: gumroad/views/admin.php (hunk 9 above)
Comment (karloscarweber) [reply gumroad/views/admin.php]: Does this mean to remove the old, before 4.9.2 documentation too?

File: gumroad/includes/misc-functions.php [hunk 13]
```diff
@@ -8,29 +8,21 @@
...
-
 	$url = 'https://gum.co/' . $args['id'];
-	$wanted = '';
-
-	if ($args['wanted'] == 'true') {
-		$wanted = '?wanted=true'
-	}
-
-	return '<a href="' . $url . $wanted . '" '. 'class="gumroad-button ' . esc_attr($args['class']) . '">' . esc_html($args['text']) . '</a>';
+	return '<a href="' . esc_url( add_query_arg( array( 'wanted' => esc_attr( $args['wanted'] )), $url ) ) . '" ' .	'class="gumroad-button ' . esc_attr($args['class']) . '">' . esc_html($args['text']) . '</a>';
```
[the remark this answers is not in the document]
Comment (karloscarweber) [reply gumroad/includes/misc-functions.php]: I actually changed this back to what it was before, it was more succinct.

**Pull request merged**